Test Team Profile Badge Process

ankitguptaindia opened the pull request:
Submitting the PR for adding a Test Handbook page for Profile Badge details. 

Closes: https://github.com/WordPress/test-handbook/issues/19

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+- Contributed to a [Test handbook](https://github.com/wordpress/test-handbook)  PR that has been merged.
+- Participated in ZIP package builds or default theme tests during a #core <release-party>.
+- Participated in a Contributor Day (as a table lead, providing technical support, or assisting with onboarding).
+- Suggested a thoughtful idea aimed at improving the quality of WordPress or enhancing the testing process in any way, which has been accepted.
```
```suggestion
- Suggested a thoughtful idea aimed at improving testing processes in any way, which gets implemented.
```
I would lean toward focusing on the testing aspect here...e.g. feature/enhancement ideas are different from active testing effort. I'm not set on the language, but just sharing an idea.

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+  
+## Test Team Badge 
+ 
+If you are currently serving as a Test Team Rep or have served as a Test Lead in any major releases of WordPress, you are eligible for a Training Team Badge.
```
```suggestion
If you have served as a Test Team Rep or Test Lead in any major release of WordPress, you are eligible for the Test Team badge.
```
Some minor tweaks.

I think that there are other qualification pathways to this as well, such as regular active participation in testing, Test Chat, etc. E.g. anyone who has been active in the team for 1 year or more. The specifics of this should be discussed with the team, but what you have here seems good for v1.

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+ 
+If you are currently serving as a Test Team Rep or have served as a Test Lead in any major releases of WordPress, you are eligible for a Training Team Badge.
+
+## Requesting a profile badge
+
+If you meet the criteria above, please request a badge using the buttons below. Kindly include any links to resources that demonstrate you have met the criteria mentioned above.
+
+[Request a Badge ](https://profiles.wordpress.org/associations/test-contributor/)                            
+
+A Test Team Reps will confirm your contribution and assign you a badge. There will be a monthly review of contributions and badges will be awarded at that time. The test team will post updates on new profile badges awarded on the [Test Team blog](https://make.wordpress.org/test/). 
```
```suggestion
## Requesting a Test Team Badge

If you meet the criteria above, please request a badge using the buttons below. Kindly include any links to resources that demonstrate you have met the criteria mentioned above.

[Request a Badge](https://profiles.wordpress.org/associations/test-contributor/)

Upon request, a Test Team Rep will confirm your contribution(s) and assign you a badge. There will be a monthly review of contributions and badges will be awarded at that time. The Test Team will post updates on new profile badges awarded on the [Test Team blog](https://make.wordpress.org/test/). 
```
For section headers, these should all be "Title Case" or "Sentence case" (refer to other handbook pages for consistency...though I think the WP docs standard is "Sentence case", which I've missed in the past 😅).

Assuming the request link will be turned into a button...not sure how to represent that in Markdown, so that looks good for now.

ironprogrammer reviewed (commented):
Thanks for this PR, @ankitguptaindia! I've left some suggestions below.

ankitguptaindia commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+
+To earn a Test Contributor Badge, you must have completed at least one of the following:
+
+- Submitted a test report comprising at least 3 tickets for [testing instructions](https://make.wordpress.org/test/handbook/test-reports/testing-instructions/), [issue reproduction](https://make.wordpress.org/test/handbook/test-reports/issue-reproduction/), or [patch testing](https://make.wordpress.org/test/handbook/test-reports/patch-testing/).
```
>Has raising this bar from 1 test report to 3 been discussed and generally agreed upon with the Test Team? If so, then I would suggest linking "at least 3 tickets" to the corresponding make/test post or discussion.

Pooja _(unable to mention here)_ and I discussed this and made this decision. This make sense to discuss with Team, I am going to create a post for discussion in test team. 

>Indicating that reports could be submitted on Trac or GitHub would help clarify this as well. Don't want to forget about all the great testing happening in Gutenberg!

Updated details and added info about Gutenberg. Thanks!

ironprogrammer replied:
Sounds good! And I believe Pooja's user is @webtechpooja here in GH.

Thinking more about _where_ reports can be submitted, perhaps this should be expanded a bit more, to incorporate Test Team's ["duty of care" areas](https://make.wordpress.org/test/handbook/#duty-of-care). For instance, testing WP-CLI, default themes, etc should qualify as well. This section could be reworded to be more general and include the wider project. What do you think?

ankitguptaindia replied:
>Sounds good! And I believe Pooja's user is @webtechpooja here in GH.

@ironprogrammer I tried to tag her again but seems like related to access or something else. I am not able to tag her name here in GH  :(  

![Screenshot 2024-03-08 at 5 50 29 PM](https://github.com/WordPress/test-handbook/assets/6384299/2ebc4a1c-913a-4955-819c-13d718674c0c)

ironprogrammer replied:
Her username might not autocomplete until she's commented or committed in this repo.

ankitguptaindia commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+To earn a Test Contributor Badge, you must have completed at least one of the following:
+
+- Submitted a test report comprising at least 3 tickets for [testing instructions](https://make.wordpress.org/test/handbook/test-reports/testing-instructions/), [issue reproduction](https://make.wordpress.org/test/handbook/test-reports/issue-reproduction/), or [patch testing](https://make.wordpress.org/test/handbook/test-reports/patch-testing/).
+- Reported a new issue on the core [Trac](https://core.trac.wordpress.org/tickets/latest) that has been merged into the WordPress release.
```
This is logical and makes sense to me. 👍🏼 
For the discussion, I've created a post where we can gather feedback. We're receiving numerous direct messages about the badge from community members. I believe implementing this new badge process for now and updating/modifying the criteria in the upcoming weeks would be beneficial. What are your thoughts?

ironprogrammer replied:
Glad to see the [call for feedback](https://make.wordpress.org/test/2024/03/08/test-team-profile-badge-your-feedback-needed/) post!

Yes, I agree that it makes sense to refer those direct requests to the post for interim guidelines (and you could also ask for their feedback/comments 😉).

ironprogrammer replied:
There wasn't much by way of [comment feedback in the post](https://make.wordpress.org/test/2024/03/08/test-team-profile-badge-your-feedback-needed/), but I'm curious about other feedback received re: "reporting an issue" counting as a test contribution. This would be a new criterion, and apply to many, many tickets that were opened not as the result of testing, but because something didn't work as the reporter expected.

As noted above, if this became a new qualification option, the stipulation I would offer is that the ticket includes [detailed testing instructions](https://make.wordpress.org/test/handbook/test-reports/testing-instructions/) to facilitate repro/patch testing. It's common for issues to remain stale/untestable while awaiting `needs-testing-info` or `reporter-feedback`. Simply reporting an issue without considering this doesn't seem like "testing" to me.

_(I realize now that I [both](https://github.com/WordPress/test-handbook/pull/20#discussion_r1489922000) a) supplied a code suggestion, and b) questioned if this qualification was valid, which made it awkward to approve the update vs debate the merits of this criterion 😅 So I've unresolved this convo to further discuss if this new option should be included.)_

ankitguptaindia replied:
Thank you for the suggestion. Removing this criterion does seem like a practical decision, especially considering the challenges in determining if ticket is not a result of testing and ticket remains stale/untestable while awaiting `needs-testing-info` or `reporter-feedback` . we can proceed with removing it from the list?

webtechpooja replied:
I also agree with the point. We can remove this point. All other things look good to me. I think we are ready to approve this.

ankitguptaindia commented:
Thanks @ironprogrammer for the feedback, I have addressed all major feedback and created a post for getting feedback on these new criteria here: [Test Team Profile Badge – Your Feedback Needed!](https://make.wordpress.org/test/?p=9156). Thanks! 

cc: @webtechpooja

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+- Participated in ZIP package builds or default theme tests during a `#core <release-party>`.
+- Participated in a Contributor Day (as a table lead, providing technical support, or assisting with onboarding).
+- Suggested a thoughtful idea aimed at improving testing processes in any way, which gets implemented.
+
```
```suggestion

```
Remove blank line.

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+ 
+If you have served as a Test Team Rep or Test Lead in any major release of WordPress, you are eligible for the Test Team badge.
+
+## Requesting a Test Team Badge
```
```suggestion
## Requesting a Test Profile Badge
```
The badges are named "Test Contributor" and "Test Team", the latter which might be confusing. I've suggested this rewording to clarify that _any_ Test profile badge should be requested with these links.

ankitguptaindia replied:
Make sense, thanks!

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+
+If you meet the criteria above, please request a badge using the buttons below. Kindly include any links to resources that demonstrate you have met the criteria mentioned above.
+
+[Request a Badge](https://profiles.wordpress.org/associations/test-contributor/)
```
```suggestion
- [Request Test Contributor Badge](https://profiles.wordpress.org/associations/test-contributor/)
- [Request Test Team Badge](https://profiles.wordpress.org/associations/test-team/)
```
Added link for Test Team badge.

ankitguptaindia replied:
wow. I was not aware that we also have group of Test Team. Would you mind adding @webtechpooja and me as a admin there? thank you!

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+  
+## Test Team Badge 
+ 
+If you have served as a Test Team Rep or Test Lead in any major release of WordPress, you are eligible for the Test Team badge.
```
```suggestion
If you have served as a Test Team Rep or Test Lead in any major release of WordPress, or have provided consistent substantial contributions to the Test Team, you are eligible for the Test Team badge.
```
Added here to include long-time and consistent team members who may not have served as rep or lead, but have demonstrated leadership in the team or contributed heavily toward its goals. These could possibly be broken into separate bullets, but I wanted to emphasize that Test contributors with a lot of impact should qualify for this badge.

ironprogrammer reviewed (commented):
I've added a few additional suggestions as this page content is shaping up.

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+
+To earn a Test Contributor Badge, you must have completed at least one of the following:
+
+- Submitted a test report comprising at least 3 tickets for [testing instructions](https://make.wordpress.org/test/handbook/test-reports/testing-instructions/), [issue reproduction](https://make.wordpress.org/test/handbook/test-reports/issue-reproduction/), or [patch testing](https://make.wordpress.org/test/handbook/test-reports/patch-testing/) either on WordPress or Gutenberg. Please note that r ...
```
```suggestion
- Submitted test reports for least 3 tickets, comprising of [testing instructions](https://make.wordpress.org/test/handbook/test-reports/testing-instructions/), [issue reproduction](https://make.wordpress.org/test/handbook/test-reports/issue-reproduction/), and/or [patch testing](https://make.wordpress.org/test/handbook/test-reports/patch-testing/). Acceptable reports should be submitted to Trac or GitHub, and can apply to any WordPress project area included in the [Test Team duty of care](https://make.wordpress.org/test/handbook/#duty-of-care).
```
Clarified that 3 individual reports are needed, and that in addition to WP/GB, testing in other areas of the project are acceptable (e.g. if someone were to test an Openverse or canonical plugin patch).

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+- Submitted unit or end-to-end tests that have been merged.
+- Contributed to a [Test handbook](https://github.com/wordpress/test-handbook) PR that has been merged.
+- Participated in ZIP package builds or default theme tests during a `#core <release-party>`.
+- Participated in a Contributor Day (as a table lead, providing technical support, or assisting with onboarding).
```
```suggestion
- Participated in a [Contributor Day](https://make.wordpress.org/test/handbook/get-started-at-contributor-day/) as a table lead, providing technical support, or assisting with onboarding.
```
Removed parenthesis and added link to Test's Contributor Day page.

ironprogrammer reviewed (commented):
Okay, one more little batch.

ankitguptaindia commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+- Reported an issue on [Core Trac](https://core.trac.wordpress.org/tickets/latest) or the [Gutenberg GitHub repo](https://github.com/wordpress/gutenberg/issues) whose fix has been merged into a WordPress release.
+- Submitted unit or end-to-end tests that have been merged.
+- Contributed to a [Test handbook](https://github.com/wordpress/test-handbook) PR that has been merged.
+- Participated in ZIP package builds or default theme tests during a `#core <release-party>`.
```
@ironprogrammer, We've received feedback suggesting the removal of this criterion to award a badge on the blog post. I would like to kindly ask for your thoughts on this matter as I am also in favor of removing this. 

Considering potential negative scenarios (as it is a nature of my QA profession 😆), it's possible that mass contributions during beta testing could lead to copying and pasting results without doing the actual testing. If individuals can easily obtain a badge through this task alone, they may be less motivated to contribute in other areas of testing.

ankitguptaindia commented on test-team-badge.md:
```diff
@@ -0,0 +1,28 @@
+- Reported an issue on [Core Trac](https://core.trac.wordpress.org/tickets/latest) or the [Gutenberg GitHub repo](https://github.com/wordpress/gutenberg/issues) whose fix has been merged into a WordPress release.
+- Submitted unit or end-to-end tests that have been merged.
+- Contributed to a [Test handbook](https://github.com/wordpress/test-handbook) PR that has been merged.
+- Participated in ZIP package builds or default theme tests during a `#core <release-party>`.
```
```suggestion
```

@ironprogrammer, We've received feedback suggesting the removal of this criterion to award a badge on the blog post. I would like to kindly ask for your thoughts on this matter as I am also in favor of removing this.

Considering potential negative scenarios (as it is a nature of my QA profession 😆), it's possible that mass contributions during beta testing could lead to copying and pasting results without doing the actual testing. If individuals can easily obtain a badge through this task alone, they may be less motivated to contribute in other areas of testing.

ironprogrammer replied:
I worry that by removing this already-established method for receiving a badge that it excludes legitimate/intentional test contributions.

There are `<release-party>` attendees who show up consistently, and new attendees where it's their first time seeing our wonderful community in action (release parties are arguably the most exciting events in Make Slack). How cool is it that you can jump right in and help test the latest release of the most popular CMS in the world, and get recognized for your effort on your first day? Badge assignments recognize those individual contributions, but can also serve to encourage further contributions.

Consider the benefits of awarding and encouraging more contributions, compared with not recognizing and missing that opportunity to bring them back. The badge shouldn't assume a contributor's "good/bad" motivations, nor should it penalize someone who has less time to contribute.

I don't think this process should focus on preventing folks from "gaming the system". Anyone who only wants the badge without intent on being a long-time contributor will find a workaround -- but at the end of the day, _they are still a contributor_, regardless of how they got here or their level of engagement 🤷.

Anyhow, that's my spiel on this one.

ironprogrammer commented:
I hold on merging this, pending feedback on the drafted page from the Test team. Once the new page is published, we can wrap up this PR 👍🏻

ankitguptaindia commented:
The Page is live now https://make.wordpress.org/test/handbook/test-team-profile-badges/  🎉

pfefferle commented:
Sorry for being late, but what about the Hosting-Test-{Runner|Reporter}:

* https://github.com/WordPress/phpunit-test-runner
* https://github.com/WordPress/phpunit-test-reporter

ironprogrammer commented on test-team-badge.md:
```diff
@@ -0,0 +1,26 @@
+To earn a Test Contributor Badge, you must have completed at least one of the following:
+
+- Submitted test reports for at least 1 ticket, comprising of [testing instructions](https://make.wordpress.org/test/handbook/test-reports/testing-instructions/), [issue reproduction](https://make.wordpress.org/test/handbook/test-reports/issue-reproduction/), and/or [patch testing](https://make.wordpress.org/test/handbook/test-reports/patch-testing/). Acceptable reports should be submitted to Tra ...
+- Submitted unit or end-to-end tests that have been merged.
```
```suggestion
- Submitted unit or end-to-end tests that have been merged.
- Contributed to a WordPress project test suite, such as [phpunit-test-runner](https://github.com/WordPress/phpunit-test-runner) or [phpunit-test-reporter](https://github.com/WordPress/phpunit-test-reporter), on a PR that has been merged.
```
Adding qualifying criteria per @pfefferle's [comment here](https://github.com/WordPress/test-handbook/pull/20#issuecomment-2076639543).

ironprogrammer reviewed (changes requested):
@ankitguptaindia As this hasn't been merged, I've added one more item here. Props @pfefferle.

ankitguptaindia commented:
> @ankitguptaindia As this hasn't been merged, I've added one more item here. Props @pfefferle.

@ironprogrammer This new criteria has been merged and added to the handbook page.  Thanks @pfefferle
